test/remote/gateways/remote_card_connect_test.rb

```diff
@@ -0,0 +1,190 @@
+
+    @amount = 100
+    @credit_card = credit_card('4788250000121443')
+    @declined_card = credit_card('4387751111111053')
+    @options = {
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+    @check = check(routing_number: '053000196')
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    options = {
+      order_id: '1',
+      ip: "127.0.0.1",
+      email: "joe@example.com",
+      po_number: '5FSD4',
+      tax_amount: '50',
+      freight_amount: '29',
+      duty_amount: '67',
+      order_date: '20170507',
+      ship_from_date: '20877',
+      items: [
+        {
+          line_no: '1',
+          material: 'MATERIAL-1',
+          description: 'DESCRIPTION-1',
+          upc: 'UPC-1',
+          quantity: '1000',
+          uom: 'CS',
+          unit_cost: '900',
+          net_amnt: '150',
+          tax_amnt: '117',
+          disc_amnt: '0'
+        },
+        {
+          line_no: '2',
+          material: 'MATERIAL-2',
+          description: 'DESCRIPTION-2',
+          upc: 'UPC-1',
+          quantity: '2000',
+          uom: 'CS',
+          unit_cost: '450',
+          net_amnt: '300',
+          tax_amnt: '117',
+          disc_amnt: '0'
+        }
+      ]
+    }
+
+    response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_successful_purchase_3DS
+    three_ds_options = @options.merge(
+      secure_flag: 'se3453',
+      secure_value: '233frdf',
+      secure_xid: '334ef34'
+    )
+    response = @gateway.purchase(@amount, @credit_card, three_ds_options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Insufficient funds', response.message
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+    assert_equal 'Approval Queued for Capture' || 'Approval Accepted', capture.message
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Insufficient funds', response.message
+  end
+
+  def test_partial_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount - 1, auth.authorization)
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@amount, '23221')
```

Comment:
Is this transaction guaranteed to never exist?

Reply:
Yea it needs to be 14 digits.

Reply:
Might be worth adding something like `@invalid_txn = '23221'` and then using that here to improve the documentation a bit then, but up to you.

Reply:
👍